### ISSUE TITLE
SA1009 handles spacing for nullable !s poorly

### ISSUE THREAD

**Issue by stephentoub** [DotNetAnalyzers/StyleCopAnalyzers#2991]
Consider:
```C#
#nullable enable
using System;

internal class Program
{
    private static void Main()
    {
    }

    private static void Foo<T>()
    {
        if (default(T)! is null)
        {
            Console.WriteLine("default(T) is null");
        }
    }
}
```

This triggers SA1009 on the:
```C#
if (default(T)! is null)
```
correcting it to be:
```C#
#nullable enable
using System;

internal class Program
{
    private static void Main()
    {
    }

    private static void Foo<T>()
    {
        if (default(T) ! is null)
        {
            Console.WriteLine("default(T) is null");
        }
    }
}
```
Such spacing is non-idiomatic and error prone.  I suggest the rule be corrected when the `!` in question is a nullable suppression.

**Comment by stephentoub** [comment]
Related to https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2968.  Not sure if it should be considered a dup.

**Comment by sharwell** [comment]
StyleCop Analyzers 1.2 will be the first release to properly support new C# 8 syntax.
